pkg/gen/additionalComments.go

```diff
@@ -164,10 +164,18 @@ func PulumiComment(kind string) string {
 }
 
 func ApiVersionComment(gvk schema.GroupVersionKind) string {
-	const template = `%s is deprecated by %s and not supported by Kubernetes v%v+ clusters.
-
-`
+	const deprecatedTemplate = `%s is deprecated by %s`
+	const notSupportedTemplate = ` and not supported by Kubernetes v%v+ clusters.`
 	gvkStr := gvk.GroupVersion().String() + "/" + gvk.Kind
 	removedIn := kinds.RemovedInVersion(gvk)
-	return fmt.Sprintf(template, gvkStr, kinds.SuggestedApiVersion(gvk), removedIn)
+
+	comment := fmt.Sprintf(deprecatedTemplate, gvkStr, kinds.SuggestedApiVersion(gvk))
+	if removedIn != nil {
+		comment += fmt.Sprintf(notSupportedTemplate, removedIn)
+	} else {
+		comment += "."
+	}
+	comment += "\n\n"
```

Comment:
Note: I removed the trailing newlines in https://github.com/pulumi/pulumi-kubernetes/pull/942/files#diff-8e79ebc2d540f3681d4e0b6bcf3a595bL167-R167, so depending on which PR is merged first, we'll need to make sure to do the same when addressing the merge conflict.